### PR TITLE
[CSS] Scoped combinators

### PR DESCRIPTION
**Pull request by Briles**
The combinators `>>>`, `>>`, `>`, `~`, `+` are now scoped

### CODE REVIEW

File: CSS/CSS.sublime-syntax [hunk 1]
```diff
@@ -725,6 +727,11 @@ contexts:
             1: punctuation.definition.entity.css
         - match: \*
           scope: entity.name.tag.wildcard.css
+          # Combinators
+          # https://drafts.csswg.org/selectors-4/#combinators
+          # https://drafts.csswg.org/css-scoping/#deep-combinator
+        - match: '({{combinators}})(?!{{combinators}})'
```
Comment (wbond) [inline CSS/CSS.sublime-syntax]: Do we need a negative lookahead here?

Comment (Briles) [inline CSS/CSS.sublime-syntax]: The negative lookahead is to ensure consecutive combinators don't match: `++` etc.

Comment (wbond) [inline CSS/CSS.sublime-syntax]: Won't that just result in the first `+` being unscoped? Perhaps instead we should match any further combinators and mark them as illegal?

**Comment by Briles** [comment]
I updated it to match invalid combinator combinations

**Comment by wbond** [comment]
Awesome, thanks!

**Comment by wbond** [comment]
I was just about to merge this, however I am wondering about the scope name, and existing color schemes. What about using `punctuation.combinator` or `punctuation.separator.combinator`? There will probably already be fairly high-contrast/color symbols to the left and right of these due to tag, class and ID names. By making it `keyword.operator`, a lot of existing packages have that kind of high-contract/color also.

**Comment by Briles** [comment]
I changed the scope to `punctuation.separator.combinator.css` as this offers the most flexibility for color scheme writers.

**Pull request merged**